### ISSUE TITLE
Bad branch count with lateinit var and anonymous function

### ISSUE THREAD

**Issue by gmousset** [Kotlin/kotlinx-kover#294]
**Describe the bug**
Hello, 

I have a problem with branch count with lateinit properties in specific conditions. I wrote a dummy example that reproduce the issue below.

Version: 0.6.1

Considering getTrucAndExecV1 method.
If the service2 property is not lateinit, there's no branche and coverage is 100%.
If service2 is lateinit, there's 1 missing branch.

I try with getTrucAndExecV2 method that do the same thing, but service2 is used from outside the anonymous function and in this case, lateinit or not, there's no branch and cover is 100%.

```
class DummyService {

    private lateinit var service2: DummyService2
//    private var service2 = DummyService2()    // lateinit or not

    fun getTrucAndExecV1(
        trucId: String
    ): String {
        service2 = DummyService2()
        return this.findTruc(trucId = trucId) { truc ->
            val t = "Hello $truc"
            service2.doDummyOperation(1, 2)
            return@findTruc t
        }
    }

    fun getTrucAndExecV2(
        trucId: String
    ): String {
        service2 = DummyService2()
        val truc = this.findTruc(trucId = trucId) { truc -> truc }
        val t = "Hello $truc"
        service2.doDummyOperation(1, 2)
        return t
    }


    private fun <T> findTruc(
        trucId: String,
        handler: (String) -> T
    ): T {
        return handler("i'm truc")
    }
}
```

Dummy test class: 
```
class DummyServiceTest {

    private var service = DummyService()

    @Test
    fun `test getTrucAndExec`() {
        val result1 = service.getTrucAndExecV1("001")
        val result2 = service.getTrucAndExecV2("001")
        Assertions.assertNotNull(result1)
        Assertions.assertNotNull(result2)
    }
}
```

![image](https://user-images.githubusercontent.com/766129/212331634-89925e9c-28f2-4531-b887-b8d8de80c332.png)

**Comment by zuevmaxim** [comment]
@gmousset Thank you for a reproducer! Please try a fix with `1.0.692` engine version:
for Kotlin Gradle script
```
kover {
    engine.set(kotlinx.kover.api.IntellijEngine("1.0.692"))
}
```
for Groovy
```
kover {
    engine = new kotlinx.kover.api.IntellijEngine("1.0.692")
}
```

**Comment by gmousset** [comment]
It works! Thank you :)

**Comment by shanshin** [comment]
Fixed in `0.7.0-Alpha`
